4/README.md

```diff
@@ -1,20 +1,41 @@
-Docker image for Varnish Cache 4.0 HTTP reverse proxy
+Varnish Cache 4.0 HTTP reverse proxy Docker image
 =====================================================
 
-This repository contains the source for Varnish Cache 4.0 Docker image.
+This container image includes Varnish 4.0 server and a reverse proxy for OpenShift and general usage.
+Users can choose between RHEL and CentOS based images.
+The RHEL image is available in the [Red Hat Container Catalog](https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/varnish-4-rhel7)
+as registry.access.redhat.com/rhscl/varnish-4-rhel7.
+The CentOS image is then available on [Docker Hub](https://hub.docker.com/r/centos/varnish-4-centos7/)
+as centos/varnish-4-centos7.
+
+
+Description
+-----------
+
+Varnish available as docker container is a base platform for 
+running Varnish server or building Varnish-based application. 
+Varnish Cache stores web pages in memory so web servers don't have to create 
+the same web page over and over again. Varnish Cache serves pages much faster 
+than any application server, giving the website a significant speed up.
+
 The image can be used as a base image for other applications based on Varnish Cache 4.0 or using s2i tool.
 
 
+Usage
+-----
+
+To build a simple [sample-app](https://github.com/sclorg/varnish-container/tree/master/4/test/test-app) application
+using standalone [S2I](https://github.com/openshift/source-to-image) and then run the
+resulting image with [Docker](http://docker.io) execute:
+
 *  **For RHEL based image**
     ```
-    $ docker pull registry.access.redhat.com/rhscl/varnish-4-rhel7
```

Comment:
Well s2i will try to pull it by itself but by default docker is only looking on hub.docker.com for images, not on registry.access.redhat.com (unless configured otherwise of course).

It makes sense to me to not mention a `docker pull` command (since the user can get the image via other means) but it would make sense to at least add something similar to mongodb's:
`For this, we will assume that you are using the rhscl/mongodb-32-rhel7 image.`